src/main/java/com/hierynomus/smbj/io/BufferedInputStreamReader.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C)2016 - SMBJ Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hierynomus.smbj.io;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+
+import com.hierynomus.smbj.common.SMBRuntimeException;
+
+public class BufferedInputStreamReader {
+    private static final int EOF = -1;
+
+    private final BufferedInputStream inputStream;
+    private volatile int cachedByte = EOF;
+
+    public BufferedInputStreamReader(BufferedInputStream inputStream) {
```

Comment:
Does it really requires a `BufferedInputStream` here? If not then a plain `InputStream` would make this clas more reusable